### ISSUE TITLE
Incompatible with Webhooks plugin

### ISSUE THREAD

**Issue by johndwells** [engram-design/SuperTable#322]
We've been using Super Table fine for a while now and have been keeping things up to date, no issues. Today we installed the Webhooks plugin (version 2.2.0), and when attempting to create a new webhook, we get a generic flash error and when we look into the console, we see a 500 error when hitting the action, `index.php?p=admin/actions/webhooks/webhooks/class-suggestions`.

The exception being thrown is due to various classes it cannot find. For example anything from the Nerds & Company Schematic plugin, or the Sprout plugins... digging into what that `class-suggestions` action is doing, it's attempting to build a complete tree of available classes to auto-complete in the CP.

As it turns out, these various 3rd party plugins (which we do not have installed) are being referenced by Super Table. Now I can see why Super Table might need to reference these 3rd party plugins for compatibility, but the upshot is that I believe I'd have to `composer require` all of those possible 3rd party plugins, just to get auto-completion in the CP for Webhooks.

To be fair, I don't think this is a Super Table issue, it's probably something better resolved from Webhooks (perhaps a settings that explicitly includes/excludes what plugins to autocomplete?). But, I thought I'd at least start the conversation here. Then if you agreed it was a Webhooks problem to solve, you could open an issue, and reference this ticket (and me too if you'd like!).

Cheers,
John

**Comment by edolyne** [comment]
I'm also experiencing similar issues with the webhooks plugin. My error states `PHP Fatal error:  Class 'verbb\supertable\errors\ElementNotFoundException' not found in /vendor/verbb/super-table/src/errors/SuperTableBlockNotFoundException.php on line 4`

**Comment by engram-design** [comment]
@edolyne That should be fixed.

@johndwells Yeah, that's an interesting thing. For instance, I've fixed one issue, but now get 

```
'barrelstrength\sproutbase\app\import\base\FieldImporter' not found in super-table/src/integrations/sproutimport/importers/fields/SuperTableImporter.php on line 10
```

Which is obviously there to add support for Sprout Importer, but only if you have that installed. I might have a chat with @brandonkelly about it further. I do think this warrants an issue on the Webhooks plugin, because while I can certainly remove those classes, that's not really desirable.

**Comment by engram-design** [comment]
Tricky, so its because it `require`'s each file, which can't seem to be error-catched. The issue isn't the file itself is missing, but when its loaded, its trying to identify all declared classes, most of which will be missing.

**Comment by johndwells** [comment]
@engram-design Yeah, that's exactly what I found in my tests... hopefully Brandon has a clever idea here.

FWIW, since I had to get around this somehow, I hacked the Webhooks plugin at the point where it's "crawling" the plugins to include more and more classes. I added code that checked if it had come upon the ST plugin and if so, just skipped over it.

So one possible Webhooks solution would be to blacklist an entire plugin... Alternatively maybe blacklist just a namespace?

Yeah, tricky indeed...

**Comment by engram-design** [comment]
Yeah, I'm surprised no one has picked up on this before, or with any other plugin for that matter. I wouldn't think this to be Super Table specific! Will have a further think...

**Comment by brandonkelly** [comment]
Interesting. Not really sure how Webhooks itself could solve this, besides limiting the suggestions to the known core classes, which isn’t ideal.

Considering this hasn’t come up until now, I suspect it’s not a very widespread problem, so maybe fixing it in Super Table is the best route? You could work around it by adding this to the top of classes that rely on third party plugins:

```php
if (!class_exists(FieldImporter::class)) {
    return;
}
```

**Comment by engram-design** [comment]
@brandonkelly Fair enough!

Fixed in [2.3.3](https://github.com/verbb/super-table/releases/tag/2.3.3)

**Comment by johndwells** [comment]
Nice! I'm glad to see it wasn't a huge refector. Thanks @engram-design!
